rllib/models/specs/specs_dict.py

```diff
@@ -1,42 +1,45 @@
+import functools
 from typing import Union, Type, Mapping, Any
 
 from ray.rllib.utils.annotations import override
 from ray.rllib.utils.nested_dict import NestedDict
-from ray.rllib.models.specs.specs_base import TensorSpecs
+from ray.rllib.models.specs.specs_base import TensorSpec
 
 
 _MISSING_KEYS_FROM_SPEC = (
     "The data dict does not match the model specs. Keys {} are "
-    "in the data dict but not on the given spec dict, and exact_match is set to True."
+    "in the data dict but not on the given spec dict, and exact_match is set to True"
 )
 _MISSING_KEYS_FROM_DATA = (
     "The data dict does not match the model specs. Keys {} are "
-    "in the spec dict but not on the data dict."
+    "in the spec dict but not on the data dict. Data keys are {}"
 )
 _TYPE_MISMATCH = (
     "The data does not match the spec. The data element "
     "{} has type {} (expected type {})."
 )
 
-SPEC_LEAF_TYPE = Union[Type, TensorSpecs]
+SPEC_LEAF_TYPE = Union[Type, TensorSpec]
 DATA_TYPE = Union[NestedDict[Any], Mapping[str, Any]]
 
+IS_NOT_PROPERTY = "Spec {} must be a property of the class {}."
 
-class ModelSpecDict(NestedDict[SPEC_LEAF_TYPE]):
-    """A NestedDict containing `TensorSpecs` and `Types`.
+
+class ModelSpec(NestedDict[SPEC_LEAF_TYPE]):
+    """A NestedDict containing `TensorSpec` and `Types`.
 
     It can be used to validate an incoming data against a nested dictionary of specs.
 
     Examples:
 
         Basic validation:
         -----------------
-        >>> spec_dict = ModelSpecDict({
+        >>> spec_dict = ModelSpec({
         ...     "obs": {
-        ...         "arm":      TensorSpecs("b, d_a", d_a=64),
-        ...         "gripper":  TensorSpecs("b, d_g", d_g=12)
+        ...         "arm":      TensorSpec("b, d_a", d_a=64),
+        ...         "gripper":  TensorSpec("b, d_g", d_g=12)
         ...     },
-        ...     "action": TensorSpecs("b, d_a", h=12),
+        ...     "action": TensorSpec("b, d_a", h=12),
         ...     "action_dist": torch.distributions.Categorical
```

Comment:
I have a random question. do you intend for these specs to be checkpointed and restored?
if so, maybe a registry of distribution is a good idea.

Reply:
I see. This is a good point. This should be incorporated into the RLModule PR then. I'll look into this there.